Formula/lua-language-server.rb

```diff
@@ -0,0 +1,103 @@
+class LuaLanguageServer < Formula
+  desc "Language Server for the Lua language"
+  homepage "https://github.com/sumneko/lua-language-server"
+  # pull from git tag to get submodules
+  url "https://github.com/sumneko/lua-language-server.git",
+      tag:      "2.5.2",
+      revision: "cb2042160865589b5534a6bf0b6c366ae4ab1d99"
+  license "MIT"
+  head "https://github.com/sumneko/lua-language-server.git", branch: "master"
+
+  depends_on "ninja" => :build
+
+  # Disable `filesystem.test_appdata_path`.
+  #
+  # This test expeccts the application to run under the user's ${HOME},
+  # which is not compatible with homebrew's build environment.
+  # See https://github.com/actboy168/bee.lua/issues/21
+  # Adapted from https://github.com/macports/macports-ports/tree/master/lua/lua-language-server
+  patch :DATA
+
+  def install
+    system "git", "submodule", "update", "--init", "--recursive"
+    chdir "3rd/luamake" do
+      system "compile/install.sh"
```

Comment:
This seems like it should be it's own formula?

Reply:
same.

Reply:
Also, `luamake` doesn't have releases, and has breaking changes relatively often in my experience. The `lua-language-server` can't be expected to build with other commits than the one currently pinned in the repo. (It's not really an independent build engine like `cmake` but rather a "build library" for embedding. Lua is big on embedding...)

Reply:
(Not that these aren't reasonable questions, mind you.)